### DISCUSSION TITLE
Image proxy server

### DISCUSSION THREAD

**Discussion by owochle** [Mailu/Mailu#3334] [Feature requests / Ideas]
It would be nice to have an image proxy server (like [imgproxy](https://github.com/imgproxy/imgproxy)) in mailu distribution, it would be safer to load images as it would not leak client IPs/infos and would allow image resizing/caching.

**Comment by nextgens** [comment]
Do you mean re-invent https://github.com/evandrofisico/roundcube-imageproxy or are you talking about something else?

**Comment by owochle** [comment]
I was thinking of something similar, but I can't find a way to add external
plugins to mailu's roundcube. Do you have a tutorial/docs?

**Reply by nextgens** [reply]
RTFM
https://mailu.io/2024.06/configuration.html#webmail-settings

**Comment by owochle** [comment]
After further reading, i was thinking more about a server-side way in order for it to work even when requesting over SMTP

**Reply by nextgens** [reply]
Then spec it out. There are lots of concerns with that type of milters starting with it can't do anything until the content has been fetched
  You probably don't want it to fetch/follow every link/images in all emails (if only because that enables open-tracking), ... And this is only the tip of the iceberg.
